src/autocomplete/shared/tokenize.ts

```diff
@@ -0,0 +1,47 @@
+import {LexerConstructor, ParserSyntaxError} from './autocomplete-types';
+import {Lexer as LexerType} from 'antlr4ng';
+import {createLexer} from './query';
+import {SqlErrorListener} from './sql-error-listener';
+import {tokenDictionary} from '../databases/yql/helpers';
+
+export type Token = {
+    ruleName: string;
+    startIndex: number;
+};
+
+export type TokenizeResult = {
+    tokens: Token[];
+    errors: ParserSyntaxError[];
+};
+
+const EOF = -1;
+
+export function tokenize<L extends LexerType>(
+    Lexer: LexerConstructor<L>,
+    symbolicNames: Array<string | null>,
+    query: string,
+): TokenizeResult {
+    const lexer = createLexer(Lexer, query);
+    const errorListener = new SqlErrorListener(tokenDictionary.SPACE);
+
+    lexer.removeErrorListeners();
+    lexer.addErrorListener(errorListener);
+
+    const tokens: Array<Token> = [];
```

Comment:
let's use `Token[]` syntax pls for consistency

Reply:
added rule https://github.com/gravity-ui/websql-autocomplete/pull/205/commits/4cf77fc3e31ca66194056f595972277fc56c4272